packages/cog/src/cli/actions/action.cog.ts

```diff
@@ -11,6 +11,33 @@ import { buildCogForQuadKey, CogJob } from '../../cog/cog';
 import { buildWarpedVrt } from '../../cog/cog.vrt';
 import { getJobPath, makeTempFolder } from '../folder';
 
+export function extractYearFromName(name: string): number {
+    const re = /(?:^|\D)(\d{4})(?:$|\D)/g;
+
+    let year = 2000;
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        const t = +m[1];
+        if (t > year) year = t;
+    }
+
+    return year;
+}
+
+export function storeImage(job: CogJob): void {
+    const now = Date.now();
+    new TileMetadataTable().create({
+        id: job.id, // FIXME - should this be calculated by name?
+        name: job.name,
+        createdAt: now,
+        updatedAt: now,
+        projection: job.projection,
+        year: extractYearFromName(job.name),
```

Comment:
Is the job.name the right place to extract the year from?

Reply:
job.name is as good as it gets.